app/models/favorites.rb

```diff
@@ -0,0 +1,15 @@
+class Favorites
+  attr_reader :contents
+
+  def initialize(initial_contents)
+    @contents = initial_contents || Hash.new
+  end
+
+  def add_pet(pet_info)
+    @contents[pet_info[:id].to_s] ||= pet_info[:info]
```

Comment:
if the pet's id isn't already in the hash, it adds that pets id as the key and its info (image and name) as the value

Reply:
so this is the big change from the work we did last night?